internal/collections/map_test.go

```diff
@@ -21,8 +21,53 @@ import (
 	"github.com/corazawaf/coraza/v3/types/variables"
 )
 
+// Case Insensitive Map
+// This is for headers and other collections that are case insensitive
 func TestMap(t *testing.T) {
-	c := NewMap(variables.ArgsPost)
+	c := NewMap(variables.RequestHeaders)
+	c.SetIndex("user", 1, "value")
+	c.Set("user-agent", []string{"value2"})
+	if c.Get("user")[0] != "value" {
+		t.Error("Error setting index")
+	}
+	if len(c.FindAll()) == 0 {
+		t.Error("Error finding all")
+	}
+	if len(c.FindString("a")) > 0 {
+		t.Error("Error should not find string")
+	}
+	if l := len(c.FindRegex(regexp.MustCompile("user.*"))); l != 2 {
+		t.Errorf("Error should find regex, got %d", l)
+	}
+
+	c.Add("user-agent", "value3")
+
+	wantStr := `REQUEST_HEADERS:
+    user: value
+    user-agent: value2,value3
```

Comment:
nit
```suggestion
    user-agent: value2; value3
```

Reply:
JC RFC 2616

> Multiple message-header fields with the same field-name MAY be present in a message if and only if the entire field-value for that header field is defined as a comma-separated list [i.e., #(values)]. It MUST be possible to combine the multiple header fields into one "field-name: field-value" pair, without changing the semantics of the message, by appending each subsequent field-value to the first, each separated by a comma. The order in which header fields with the same field-name are received is therefore significant to the interpretation of the combined field value, and thus a proxy MUST NOT change the order of these field values when a message is forwarded.

Reply:
@jcchavezs do you still think we need to change something here?

Reply:
no need